src/chunk.c

```diff
@@ -2758,15 +2759,19 @@ ts_chunk_exists_with_compression(int32 hypertable_id)
 	init_scan_by_hypertable_id(&iterator, hypertable_id);
 	ts_scanner_foreach(&iterator)
 	{
-		bool isnull_dropped;
-		bool isnull_chunk_id =
-			slot_attisnull(ts_scan_iterator_slot(&iterator), Anum_chunk_compressed_chunk_id);
+		bool isnull_dropped, status_isnull;
+		bool status_is_compressed =
+			ts_flag_is_set_32(DatumGetInt32(slot_getattr(ts_scan_iterator_slot(&iterator),
+														 Anum_chunk_status,
+														 &status_isnull)),
+							  CHUNK_STATUS_COMPRESSED);
+		Assert(!status_isnull);
 		bool dropped = DatumGetBool(
 			slot_getattr(ts_scan_iterator_slot(&iterator), Anum_chunk_dropped, &isnull_dropped));
 		/* dropped is not NULLABLE */
 		Assert(!isnull_dropped);
 
-		if (!isnull_chunk_id && !dropped)
+		if (status_is_compressed && !dropped)
```

Comment:
You are changing the semantics of this function. This is used to test if a corresponding compressed chunk exists for a chunk. If you need this new functionality it would be better to add a new function that clearly checks for status.

Reply:
Probably should also add a  comment that this function checks for the presence of a  corresponding compressed chunk